src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Reflection;
+
+    internal class HeartbeatDefaultPayload
+    {
+        public const string UpdatedFieldsPropertyKey = "updatedFields";
+
+        public static readonly string[] DefaultFields =
+        {
+            "runtimeFramework",
+            "baseSdkTargetFramework",
+            UpdatedFieldsPropertyKey
+        };
+
+        private List<string> enabledProperties;
+
+        public HeartbeatDefaultPayload() : this(null)
+        {
+        }
+
+        public HeartbeatDefaultPayload(IEnumerable<string> disableFields)
+        {
+            this.SetEnabledProperties(disableFields);
+        }
+
+        public IDictionary<string, HeartbeatPropertyPayload> GetPayloadProperties()
+        {
+            var payload = new Dictionary<string, HeartbeatPropertyPayload>();
+            foreach (string fieldName in this.enabledProperties)
+            {
+                switch (fieldName)
+                {
+                    case "runtimeFramework":
+                        payload.Add(fieldName, new HeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetRuntimeFrameworkVer()
+                        });
+                        break;
+                    case "baseSdkTargetFramework":
+                        payload.Add(fieldName, new HeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
```

Comment:
thinking out loud here.. so when does is-healthy becomes false ever?

Reply:
The 'isHealthy' field is to allow extensions that add their own properties to the heartbeat payload to denote an 'unhealthy' state. It is up to the caller to set this true/false. Once we get in all of the default payload items, there may be one or two that we will set with this set to false as well.